sklearn/multioutput.py

```diff
@@ -0,0 +1,121 @@
+    """Multi target regression
+
+    This strategy consits of fitting one regressor per target. This is a
+    simple strategy for extending regressors that do not natively support
+    multi-target regression.
+
+    Parameters
+    ----------
+    estimator : estimator object
+        An estimator object implementing `fit` and `predict`.
+
+    n_jobs : int, optional, default: 1
+        The number of jobs to run in parallel for `fit`. If -1,
+        then the number of jobs is set to the number of cores.
+    """
+    def __init__(self, estimator, n_jobs=1):
+        self.estimator = estimator
+        self.n_jobs = n_jobs
+
+    def fit(self, X, y, sample_weight=None):
+        """Fit underlying regressors
+
+        Parameters
+        ----------
+        X : array-like, shape = [n_samples, n_features]
+            Data.
+
+        y : array-like, shape = [n_samples, n_targets]
+            The multiple target values.
+
+        sample_weight : array-like, shape = [n_samples] or None
+            Sample weights. If None, then samples are equally weighted.
+            Only supported if the underlying regressor supports sample
+            weights.
+
+        Returns
+        -------
+        self
+        """
+        X, y = check_X_y(X, y,
+                         multi_output=True,
+                         accept_sparse=['csr', 'csc', 'coo', 'dok', 'lil'])
+
+        if y.ndim == 1:
+            raise ValueError("y must have at least two dimensions for "
+                             "multi target regression but has only one.")
+
+        if (sample_weight is not None and
+            not has_fit_parameter(self.estimator, 'sample_weight')):
+            raise ValueError("Underlying regressor does not support sample weights.")
+
+        self.estimators_ = Parallel(n_jobs=self.n_jobs)(delayed(_fit_regression)(
+            self.estimator, X, y[:, i]) for i in range(y.shape[1]))
+        return self
+
+    def predict(self, X, sample_weight=None):
+        """Predict regression values for X.
+
+        Parameters
+        ----------
+        X : array-like, shape = [n_samples, n_features]
+            Data.
+
+        sample_weight : array-like, shape = [n_samples] or None
+            Sample weights. If None, then samples are equally weighted.
+            Only supported if the underlying regressor supports sample
+            weights.
+
+        Returns
+        -------
+        y : array of shape [n_samples, n_targets]
+            Predicted values.
+        """
+        check_is_fitted(self, 'estimators_')
+
+        X = check_array(X, accept_sparse=['csr', 'csc', 'coo', 'dok', 'lil'])
+
+        if (sample_weight is not None and
+            not has_fit_parameter(self.estimator, 'sample_weight')):
+            raise ValueError("Underlying regressor does not support sample weights.")
+
+        pred = Parallel(n_jobs=self.n_jobs)(delayed(_parallel_helper)(e, 'predict', X)
```

Comment:
Doing this in parallel might not be beneficial for algorithms where prediction is quick for instance linear models due to the joblib overhead. It should be documented